src/lib/wallet/GoodWalletProvider.js

```diff
@@ -285,7 +287,10 @@ export const GoodWalletProvider = ({ children, disableLoginAndWatch = false }) =
         await goodWallet.setIsPollEvents(false) //stop watching prev chain events
         await goodWallet.init({ network: contractsNetwork }) //reinit wallet
         let web3Provider = new Web3Provider(
-          new PrivateKeyProvider(goodWallet.wallet.eth.accounts.wallet[0].privateKey, goodWallet.wallet._provider.host),
+          new JsonRpcProviderWithSigner(
```

Comment:
Bridging only works on initial loaded chain, stop working after switching

Shouldn't this also have the extra wrapper for the multhttpProvider?

Reply:
then I think that issue should be taken up within this PR
as currently if I load the wallet, bridging works
after switching chain, it stops working and you get stuck on the confirm screen of bridge as the request never gets send